extensions/wikia/Editcount/SpecialEditcount_body.php

```diff
@@ -198,27 +198,19 @@ function editsByNsAll( $uid ) {
 
 		if ( empty($nscount) ) {
 			$nscount = array();
-			if ( !empty( $wgStatsDBEnabled ) ) {
-				$dbs = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
-				$res = $dbs->select(
-					array( 'events' ),
-					array( 'page_ns as namespace', 'count(page_ns) as count' ),
-					array(
-						'user_id' => $uid,
-						' ( event_type = 1 ) or ( event_type = 2 ) '
-					),
-					__METHOD__,
-					array (
-						'GROUP BY' => 'page_ns',
-						'ORDER BY' => 'null'
-					)
-				);
+			$dbs = wfGetDB(DB_SLAVE, array(), $wgDWStatsDB);
+			$res = $dbs->select(
+				array( 'rollup_wiki_namespace_user_events' ),
+				array( 'namespace_id as namespace', 'sum(creates + edits) as count' ),
+				array( 'user_id' => $uid, 'period_id' => DataMartService::PERIOD_ID_DAILY ),
+				__METHOD__,
+				array ( 'GROUP BY' => 'namespace_id' )
+			);
```

Comment:
Let's make sure that indices are used by this query

Reply:
```mysql@geo-db-dataware-slave.query.consul[statsdb]>describe select sum(creates + edits) from statsdb.rollup_wiki_namespace_user_events where user_id = 455627 and period_id = 1;
+----+-------------+-----------------------------------+------+------------------+------------------+---------+-------------+------+-------+
| id | select_type | table                             | type | possible_keys    | key              | key_len | ref         | rows | Extra |
+----+-------------+-----------------------------------+------+------------------+------------------+---------+-------------+------+-------+
|  1 | SIMPLE      | rollup_wiki_namespace_user_events | ref  | user_period_time | user_period_time | 6       | const,const |  617 | NULL  |
+----+-------------+-----------------------------------+------+------------------+------------------+---------+-------------+------+-------+```

REF - I'm assuming that's good?

Reply:
`Using temporary; Using filesort` is fine as long as it's not performed on thousands of rows.